### PR TITLE
Fix wc_get_price_excluding_tax ignoring the woocommerce_adjust_non_base_location_prices filter.

### LINKED ISSUE #31721: woocommerce_adjust_non_base_location_prices not changing product price on checkout since v5.9.0
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

With certain tax settings to have the prices be inclusive of VAT, regardless of the VAT percentage of the customer's country, the checkout cart has prices listed for the excl. VAT amount using the store's VAT percentage, not the costumer's. Thus, the prices of the products do not add up to the subtotal. This presents a very confusing cart (see screenshot under steps to reproduce).

I noticed this bug after updating WooCommerce from 5.5.2 to 5.9.0 and this bug is still present in WooCommerce 6.1.1 (see steps to reproduce).

A workaround for this issue is to set "Display prices during cart and checkout" to "Including tax". This makes the product price to be the price including tax and the VAT is listed in brackets after the total. Not a pretty solution in my opinion.

### Expected behavior

When a country is selected with a VAT rate other than the store's home country, the product prices should be updated to show their new price excluding VAT so that they add up to the displayed subtotal.

### Actual behavior

When a country is selected with a VAT rate other than the store's home country, the product prices show the price they would be in the store's home country, while the subtotal is updated to the correct amount. Thus, the total of all the products will be different than the displayed subtotal, which is confusing.

### Steps to reproduce

The following has been tested with a fresh install of Wordpress 5.8.3 and WooCommerce 6.1.1, using the default Twenty Twenty-One theme.

1. Add to the functions.php of the theme: `add_filter( 'woocommerce_adjust_non_base_location_prices' , '__return_false' );`
2. Enable taxes in WooCommerce and use the following settings for taxes:
image
3. Add tax rates by country (tested with [this CSV](https://woocommerce.com/wp-content/uploads/2021/01/vat_rates.csv) from [the EU VAT Rates guide](https://woocommerce.com/document/setting-up-eu-vat-rates-for-digital-products/)).
4. Set the store's country to one of the countries you just added a tax rate for (tested with store being based in the Netherlands).
5. Create a product, add it to the cart, and navigate to checkout.
6. On the checkout page, change the country to any country with a VAT rate that is different from the store's home country (tested with Germany, Ireland, Hungary) and see that the product price and subtotal are mismatched.
image

### WordPress Environment ###

WordPress address (URL): https://local.******.com
Site address (URL): https://local.******.com
WC Version: 6.1.1
REST API Version: ✔ 6.1.1
WC Blocks Version: ✔ 6.5.2
Action Scheduler Version: ✔ 3.4.0
WC Admin Version: ✔ 3.0.3
Log Directory Writable: ✔
WP Version: 5.8.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.1
PHP Version: 7.4.9
PHP Post Max Size: 32 MB
PHP Time Limit: 400
PHP Max Input Vars: 10000
cURL Version: 7.54.0
OpenSSL/1.0.2u

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.29-MariaDB
Max Upload Size: 32 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ❌ Your server does not support the gzopen function - this is required to use the GeoIP database from MaxMind.
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.1.1
WC Database Prefix: wp_
Total Database Size: 3.79MB
Database Data Size: 2.39MB
Database Index Size: 1.40MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 1.50MB + Index: 0.03MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 2
page: 7
post: 2
product: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 6.1.1

### Inactive Plugins (2) ###

Akismet Anti-Spam: by Automattic – 4.2.1
Hello Dolly: by Matt Mullenweg – 1.7.2

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: left_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /?page_id=6
Cart: #7 - /?page_id=7
Checkout: #8 - /?page_id=8
My account: #9 - /?page_id=9
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-One
Version: 1.4
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 5
Oldest: 2022-01-25 10:50:43 +0000
Newest: 2022-01-25 10:50:42 +0000

### Status report information ###

Generated at: 2022-01-25 11:24:35 +00:00

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by oscargare**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

The commit [#d98112e](https://github.com/woocommerce/woocommerce/commit/d98112e0140724a9ad5c00991211a1901e7b7525) broke the `wc_get_price_excluding_tax` function when the `woocommerce_adjust_non_base_location_prices` filter returns false.

The function now only removes the taxes of the customer location if an order is passed as a parameter, ignoring the `woocommerce_adjust_non_base_location_prices` filter.

The issue causes that for the configuration: Price entered with tax, Display price in the shop "exclude tax," and `woocommerce_adjust_non_base_location_prices` set to false, the shop displays incorrect prices.

This PR fixes the issue by setting the customer to null when no order is passed as a parameter. 
Fix #31721 
The PR also keep the functionality introduced in the PR [#30692](https://github.com/woocommerce/woocommerce/pull/30692)

### How to test the changes in this Pull Request:

1. Follow the steps of #31721
2. Follow the steps of the PR [#30692](https://github.com/woocommerce/woocommerce/pull/30692)

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Fixed: wc_get_price_excluding_tax does not remove the customer location taxes when the filter woocommerce_adjust_non_base_location_prices returns false.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by oscargare** [comment]
@Konamiman Could you review the Pull to include this fix in the next version? thanks

**Comment by Konamiman** [comment]
Hi @oscargare, thanks for your contribution. Your change looks good, but it breaks a unit test that will need to be fixed before your pull request can be merged. I can do that myself if you set "Allow edits from maintainers" in your fork (guide here: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)

**Comment by oscargare** [comment]
Hi @Konamiman , the option "Allow edits by maintainers" is cheked.

**Review by Konamiman — APPROVED** [review]

**Comment by Konamiman** [comment]
Ok, I must have done something wrong the first time, now I was able to introduce the changes needed. Approved and merged, thank you!

**Comment by oscargare** [comment]
Great! :+1:

**Pull request merged**
